### ISSUE TITLE
Decorations can spawn on dungeons

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#12427]
##### Minetest version
```
5.5.0
```
##### Summary
Decorations like grass, plants, flowers, etc. can sometimes appear on dungeons when they appear on the surface level. Floating plants are also possible.

https://www.satoshiupload.com/images/s1Sxddlace.png

##### Steps to reproduce

1. Jack up the dungeon spawn rate way up
2. Create a DevTest world
3. Activate the following mod
4. Start the world, fly around and look at dungeons at the surface

Mod code:
```
minetest.clear_registered_decorations()
minetest.register_decoration({
	deco_type = "simple",
	fill_ratio = 10.0,
	decoration = {"basenodes:junglegrass"},
	sidelen = 16,
	place_on = {"basenodes:dirt_with_grass", "basenodes:dirt", "basenodes:sand"},
	y_min = -31000,
	y_max = 31000,
})
```

(**Important! This mod must depend on `basenodes` and `mapgen`**)

* Expected result: Jungle grass only spawns on dirt, dirt-with-grass and sand.
* Actual result: Jungle grass spawns on dirt, dirt-with-grass and sand, but sometimes also on the dungeons or in air.

##### Coordinates

Here's some example mapgen setting and coordinates to reproduce:

Settings:
```
mgv7_np_dungeons = {
	flags = 
	lacunarity = 2
	persistence = 0.8
	seed = 0
	spread = (50,50,50)
	scale = 0.5
	octaves = 2
	offset = 50
}
```

* Mapgen: v7
* Seed: 5510618407724436858
* Floating plants coordinates: -980, 13, 136
* Plants on cobblestone coordinates: -282, 25, -663

I'm not sure if these coordinates will always work. If not, just try to find your own bad plants, it should be relatively easy to find.

**Comment by Wuzzy2** [comment]
I have a guess: Dungeons sometimes "grow" outside of mapblock boundaries / into `ignore`, thus leading to all sorts of bugs, including this one. This would explain why it only happens sometimes.

**Comment by IrrlichtMt4** [comment]
I think it's better anyways with plants growing on dungeons. It makes the dungeon look ancient.

**Comment by IrrlichtMt4** [comment]
Can confirm bug though.

**Comment by Wuzzy2** [comment]
Yes, plants on dungeons might be nice, but that's missing the point. The game needs to have full control over where decorations grow. Growing on a node not specified in `place_on` is a bug. Growing on air is a bug.

**Comment by Zughy** [comment]
I couldn't find anything at your coordinates, but I've found this nearby (-305, 20, -524)

![imagen](https://user-images.githubusercontent.com/63455151/174171404-4f315ab8-f97f-4c06-8723-0ede013d3a67.png)
